back/engines/commercial/bulk_import_ideas/app/controllers/bulk_import_ideas/web_api/v1/bulk_import_controller.rb

```diff
@@ -0,0 +1,169 @@
+        },
+        'pdf' => {
+          exporter_class: Exporters::IdeaPdfFormExporter,
+          parser_class: Parsers::IdeaPdfFileParser
+        }
+      }
+    }
+
+    def bulk_create
+      send_not_found unless supported_model? && supported_format?
+
+      file = bulk_create_params[:file]
+      locale = params[:import] ? bulk_create_params[:locale] : current_user.locale
+      personal_data_enabled = params[:import] ? bulk_create_params[:personal_data] || false : false
+      phase_id = params[:id]
+      file_parser = file_parser_service.new(current_user, locale, phase_id, personal_data_enabled)
+      import_service = importer_service.new(current_user)
+
+      rows = file_parser.parse_file file
+      ideas = import_service.import(rows)
+      users = import_service.imported_users
+
+      sidefx.after_success current_user, @phase, params[:model], params[:format], ideas, users
+
+      render json: serializer.new(
+        ideas,
+        params: jsonapi_serializer_params,
+        include: %i[author idea_import]
+      ).serializable_hash, status: :created
+    rescue BulkImportIdeas::Error => e
+      sidefx.after_failure current_user, @phase, params[:model], params[:format]
+      render json: { errors: { file: [{ error: e.key, **e.params }] } }, status: :unprocessable_entity
+    end
+
+    def export_form
+      send_not_found unless supported_model? && supported_format?
+
+      locale = params[:locale] || current_user.locale
+      personal_data_enabled = params[:personal_data] == 'true'
+
+      service = form_exporter_service.new(@phase, locale, personal_data_enabled)
+      send_data service.export, type: service.mime_type, filename: service.filename
+    end
+
+    def approve_all
+      send_not_found unless supported_model?
+
+      records = imported_draft_records
+      approved = 0
+      not_approved = 0
+      records.each do |record|
+        if record.update(publication_status: 'published')
+          approved += 1
+        else
+          not_approved += 1
+        end
+      end
+
+      render json: raw_json({ approved: approved, notApproved: not_approved })
+    end
+
+    def draft_records
+      send_not_found unless supported_model?
+
+      render json: linked_json(
+        paginate(imported_draft_records),
+        serializer,
+        include: %i[author idea_import],
+        params: jsonapi_serializer_params
+      )
+    end
+
+    # Show the metadata of a single imported idea
+    # TODO: When other model types (eg comments) are supported, IdeaImport & IdeaImportFile will need to be made polymorphic
+    def show_idea_import
+      idea_import = IdeaImport.find(params[:id])
+      authorize idea_import
+
+      render json: WebApi::V1::IdeaImportSerializer.new(
+        idea_import,
+        params: jsonapi_serializer_params
+      ).serializable_hash
+    end
+
+    def show_idea_import_file
+      idea_import_file = IdeaImportFile.find(params[:id])
+      authorize idea_import_file.project
```

Comment:
Maybe it's not immediately obvious that one controller uses two policies `IdeaImportPolicy` and `ProjectPolicy`. Also, `ProjectPolicy` seems to become too fat. I see previously we had `ImportIdeasPolicy`. Why did you stop using it?

Not a blocker though.